theano/compile/nanguardmode.py

```diff
@@ -55,13 +60,14 @@ def flatten(l):
     return rval
 
 
-def contains_nan(arr):
+def contains_nan(arr, nd=None):
     """
     Test whether a numpy.ndarray contains any `np.nan` values.
 
     Parameters
     ----------
-    arr : np.ndarray
+    arr : np.ndarray or output of any Theano op
+    nd : if the output of an Theano op, the node associated to it
```

Comment:
This doesn't have the right format for a parameter.  The format is
```
name : type
    descr
```
and what you have for `nd` is not a type.

`nd` makes me thinks about a number of dimensions so I think `node` is a much better name.

of an Theano -> of a Theano

Finally, I think it would be better design to have contains_nan be a method of Type.  I'm not sure what the best design would be to make exceptions like for GPU_mrg_uniform, though.